src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java

```diff
@@ -316,7 +311,7 @@ public static Configuration loadConfiguration(InputSource configSource,
                     new ConfigurationLoader(overridePropsResolver,
                             omitIgnoreModules, threadModeSettings);
             loader.parseInputSource(configSource);
-            return loader.configuration;
+            return loader.saxHandler.configuration;
```

Comment:
Code looks strange, too much chain of fields. 
It is conceptually the same.
Why you think this is not false positive?

Reply:
I only looked at the message and not the place it was coming from `may expose internal representation by returning`.

This more like EI_EXPOSE_REP:
> May expose internal representation by returning reference to mutable object (EI_EXPOSE_REP)
Returning a reference to a mutable object value stored in one of the object's fields exposes the internal representation of the object.  If instances are accessed by untrusted code, and unchecked changes to the mutable object would compromise security or other important properties, you will need to do something different. Returning a new copy of the object is better approach in many situations.

It didn't seem like a false as we have a public method returning the instance of a private field, which represents the same class we are in. `ConfigurationLoader` only has `private` constructors.

I would prefer to make changes to `XmlLoader` to accomodate this better since I can't otherwise pass in and out an instance, but I was not seeing any easy way to do this.

Reply:
> void parseInputSource(InputSource source)

Method in `ConfigurationLoader` was changed to return the instance now.